### PR TITLE
Add 'History' command

### PR DESCRIPTION
**Pull request by alexeitcekov**

### CODE REVIEW

File: test/commands-tests/history.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+            let initResult = await execute(constants.cliCommands.INIT, []);
+            clearInterval(initInterval);
+    
+            console.log();
+            console.log(initResult);
+    
+            // // When master branch has history functionality this command should be removed.
+            await exec('npm', 'link', ['forgae']);
+    
+            console.log('Starting NODE..');
```
Comment (maptuhec) [inline test/commands-tests/history.js]: Remove console logs

File: cli-commands/forgae-deploy/forgae-deployer.js [hunk 2]
```diff
@@ -64,15 +118,29 @@ class Deployer {
...
+
+        const deployPromise = await compiledContract.deploy(deployOptions);
         const deployedContract = await deployPromise;
 
         let regex = new RegExp(/[\w]+.aes$/);
-        let contractFileName = regex.exec(contractPath)
+        let contractFileName = regex.exec(contractPath);
+
+        let txInfo = await getTxInfo(deployedContract.transaction, this.network);
+        const isSuccess = txInfo.gasPrice >= 0 && txInfo.gasUsed > 0;
```
Comment (Perseverance) [inline cli-commands/forgae-deploy/forgae-deployer.js]: So the transaction wont use gas if the transaction has failed? I do not feel very confident in this way of figuring out if the transaction is successful

File: cli-commands/forgae-history/log-store-service.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	 * Gets the last written record.
+	 */
+	getLastWorkingRecord() {
+		const history = this.getHistory();
+		return this._historyStore.get('' + (history.length - 1));
+	}
+
+	/**
+	 * 
+	 * Add a record to the history of deployments
```
Comment (Perseverance) [inline cli-commands/forgae-history/log-store-service.js]: Remove this comment if it is not correct

File: cli-commands/forgae-history/utils.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+//const ethersUtils = require("ethers").utils;
```
Comment (Perseverance) [inline cli-commands/forgae-history/utils.js]: Remove this if not needed

File: cli-commands/forgae-test/test.js [hunk 5]
```diff
@@ -35,7 +34,7 @@ const run = async (path) => {
   }
 
   testDirectory = `${process.cwd()}/${path}`;
-  const files = await utils.getFiles(`${process.cwd()}/${path}/`, `.*\.(js|es|es6|jsx|sol)$`);
+  const files = await utils.getFiles(`${process.cwd()}/${path}/`, `.*\\.(js|es|es6|jsx|sol)$`);
```
Comment (Perseverance) [inline cli-commands/forgae-test/test.js]: What is the change here? Also what is sol doing here?

Comment (alexeitcekov) [reply cli-commands/forgae-test/test.js]: when you write a regex in js '\' should be escaped too. if you use '\.', the back slash do not escape dot(.) and match everything

Comment (alexeitcekov) [reply cli-commands/forgae-test/test.js]: .sol was there ...

**Pull request merged**